packages/teams-js/src/public/meeting.ts

```diff
@@ -210,6 +210,39 @@ export namespace meeting {
     error?: SdkError;
   }
 
+  /**
+   * Interface for mic state change
+   *
+   * @beta
+   */
+  export interface MicState {
+    id: number;
+    /**
+     * Indicates the mute status of the mic.
+     */
+    isMicMuted: boolean;
+  }
+
+  export interface MeetingFailedResponse {
+    id: number;
+    /**
+     * The reason for the failure
+     */
+    reason: string;
+    /**
+     * This property is currently unused.
+     */
+    message?: string;
+  }
+
+  export interface RequestAppAudioHandlingParams {
+    isAppHandlingAudio: boolean;
+    callbackMicMuteStateChangedHandler: (
+      micState: MicState,
+      callbackResponse: (error?: MeetingFailedResponse, micState?: MicState) => void,
```

Comment:
I remember we said we should remove this `callbackResponse` parameter because it's a non-established pattern. Without it though, I don't think we can do the thing where the mute response is automatically sent (line 645 - 649 in this file), since muting/unmuting on the app side would be an async operation. Instead, the app would have to call the other API `updateMicState` to notify the host that the mute operation is done.

Unless there's some way to use async for this, I think we need to make it so app _have_ to call the other API to confirm the mute/unmute.